runtime/parachains/src/builder.rs

```diff
@@ -86,24 +104,40 @@ impl<T: paras_inherent::Config> BenchBuilder<T> {
 	/// of the functions in this implementation.
 	pub(crate) fn new() -> Self {
 		BenchBuilder {
-			// Validators should be declared prior to all other setup.
 			validators: None,
-			// Starting block number; we expect it to get incremented on session setup.
 			block_number: Zero::zero(),
-			// Starting session; we expect it to get incremented on session setup.
 			session: SessionIndex::from(0u32),
-			// Session we want the scenario to take place in. We roll to to this session.
 			target_session: 2u32,
-			// Optionally set the max validators per core; otherwise uses the configuration value.
 			max_validators_per_core: None,
-			// Optionally set the max validators; otherwise uses the configuration value.
 			max_validators: None,
-			// Optionally set the number of dispute statements for each candidate,
 			dispute_statements: BTreeMap::new(),
+			dispute_sessions: Default::default(),
+			backed_and_concluding_cores: Default::default(),
+			code_upgrade: None,
 			_phantom: sp_std::marker::PhantomData::<T>,
 		}
 	}
 
+	/// Set `self.dispute_sessions`.
+	pub(crate) fn set_dispute_sessions(mut self, dispute_sessions: Vec<u32>) -> Self {
+		self.dispute_sessions = dispute_sessions;
+		self
+	}
+
+	/// Set `self.backed_and_concluding_cores`.
+	pub(crate) fn set_backed_and_concluding_cores(
+		mut self,
+		backed_and_concluding_cores: BTreeMap<u32, u32>,
+	) -> Self {
+		self.backed_and_concluding_cores = backed_and_concluding_cores;
+		self
+	}
+
+	pub(crate) fn set_code_upgrade(mut self, code_upgrade: Option<u32>) -> Self {
```

Comment:
nit: use `impl Into<u32>` since `Option<T>` provides an `impl From<T> for T {..}` https://doc.rust-lang.org/std/option/enum.Option.html#impl-From%3CT%3E

Reply:
I think this is what you mean: 0dbe497